### PR TITLE
refactor(orc8r): Split state to protected and southbound 

### PR DESCRIPTION
**Pull request by christinewang5**
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]

File: orc8r/cloud/go/services/state/servicers/protected/servicer.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/thoas/go-funk"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"magma/orc8r/cloud/go/blobstore"
+	servicers "magma/orc8r/cloud/go/services/state/servicers/southbound"
+	"magma/orc8r/lib/go/protos"
+)
+
+type stateServicer struct {
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/state/servicers/protected/servicer.go]: Nit: cloudStateServicer, prob

File: orc8r/cloud/go/services/state/servicers/southbound/validate.go [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
 
 package servicers
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/state/servicers/southbound/validate.go:14]: Do the fns in this pkg need to be public? Is fine if yes, but isn't clear to me that yes

File: orc8r/cloud/go/services/state/servicers/southbound/helpers.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package servicers
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/state/servicers/southbound/helpers.go]: If we're gonna convert this file's contents to public, let's put it in like orc8r/cloud/go/services/state/conversion.go, or something like that (and maybe the err goes in .../error.go)

**Review by hcgatewood — APPROVED** [review]
Lgtm! Couple nits

**Review by hcgatewood — CHANGES_REQUESTED** [review]
Oh jk, don't forget to add the southbound servicer to the main.go! (And any test service initializations)

File: orc8r/cloud/go/services/state/state/main.go [hunk 4]
```diff
@@ -101,7 +101,7 @@ func newStateServicer(store blobstore.StoreFactory) protos.StateServiceServer {
 }
 
 func newCloudStateServicer(store blobstore.StoreFactory) protos.CloudStateServiceServer {
-	servicer, err := servicers.NewCloudStateServicer(store)
+	servicer, err := protected_servicers.NewCloudStateServicer(store)
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/state/state/main.go:104]: !

**Review by hcgatewood — APPROVED** [review]
Jk I see where this one was already correct for initializing the cloud+southbound servicers 🙂

File: orc8r/cloud/go/services/state/servicers/southbound/validate.go (hunk 2 above)
Comment (christinewang5) [reply orc8r/cloud/go/services/state/servicers/southbound/validate.go:14]: Yeah it's used in the protected ones 😕https://github.com/christinewang5/magma/blob/195f312f6005b04995ff95e09a1a0da8d0e623b7/orc8r/cloud/go/services/state/servicers/protected/servicer.go#L42-L42

File: orc8r/cloud/go/services/state/state/main.go (hunk 4 above)
Comment (christinewang5) [reply orc8r/cloud/go/services/state/state/main.go:104]: what's wrong with this one!?

**Pull request merged**